src/parser/open-api/schemaParser.js

```diff
@@ -11,127 +12,164 @@ const literalTypes = ['string', 'integer', 'number', 'boolean']
  *
  * @return {Object}
  */
-function getAdditionalPropertiesProperty (additionalProperties) {
-  const property = getPropertyNode('additionalProperties', additionalProperties)
+function getUIPropertyForAdditionalProperties (additionalProperties) {
+  const uiProperty = getUIProperty('additionalProperties', additionalProperties)
 
   // Also add some meta data to it, so the UI can handle this property differently if required
-  property.attributes = {}
-  property.attributes.isAdditionalProperties = true
+  uiProperty.attributes = {}
+  uiProperty.attributes.isAdditionalProperties = true
 
-  return property
+  return uiProperty
 }
 
 /**
- * Construct UI ready property object from given inputs.
+ * Construct a UI ready property object for circular reference. This is mainly for debugging purpose.
+ *
+ * @param {string} name
+ *
+ * @return {Object}
+ */
+function getUIPropertyForCircularReference (name) {
+  const visitedNodes = Array.from(checkedNodes.values()).join(', ')
+  const uiProperty = getBaseUIProperty('CIRCULAR REFERENCE', {})
+
+  uiProperty.description = `Circular reference found in **${name}**. Visited nodes: ${visitedNodes}`
```

Comment:
I also find the "found in" part a little confusing. It's more saying, "I have seen this schema before in "categories"".